File: 94-binary-tree-inorder-traversal/binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        st = []
        def inorder(root):
            if not root: return
            inorder(root.left) 
            st.append(root.val)
            inorder(root.right)
        inorder(root)
        return st
        
        """inOrder = []
        stack = []
        node = root
        while True:
            if node:
                stack.append(node)
                node = node.left
            else:
                if not stack: break
                else:
                    node = stack.pop()
                    inOrder.append(node.val)
                    node = node.right
        return inOrder"""

File: 94-binary-tree-inorder-traversal/binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private void inorder(TreeNode root, ArrayList<Integer> arr){
        if(root == null) return;
        this.inorder(root.left,arr);
        arr.add(root.val);
        this.inorder(root.right,arr);
    }
    public List<Integer> inorderTraversal(TreeNode root) {
        ArrayList<Integer> arr = new ArrayList<>();
        this.inorder(root, arr);
        return arr;
    }
}